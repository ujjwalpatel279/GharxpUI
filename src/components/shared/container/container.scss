// ========================================================================
// COMPONENTS/SHARED/CONTAINER
// ========================================================================

@import '~sassHelpers';

.c-container {
  @include container;

  &--ctabar {
    padding-bottom: 78px;
  }

  &--fullwidth {
    max-width: 100%;
  }
}

.c-container-fullWidth {
  margin-left: auto;
  margin-right: auto;
  padding-left: 0.5rem;
  padding-right: 00.5rem;
}
