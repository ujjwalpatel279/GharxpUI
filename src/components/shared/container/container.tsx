import React, { FunctionComponent } from 'react';

import './container.scss';

interface ContainerProps {
  extendClass?: string;
  ctabar?: boolean;
  fullWidth?: boolean;
}

export const Container: FunctionComponent<ContainerProps> = ({ children, extendClass, fullWidth, ctabar }) => {
  let containerClasses = 'c-container';

  if (extendClass) containerClasses += ` ${extendClass}`;

  if (ctabar) {
    containerClasses += ' c-container--ctabar';
  }

  if (fullWidth) {
    containerClasses += ' c-container--fullwidth';
  }

  return <div className={containerClasses}>{children}</div>;
};
