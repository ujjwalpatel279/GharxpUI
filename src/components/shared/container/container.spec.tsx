import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Container } from './container';

describe('Container Component', () => {
  afterEach(cleanup);

  it('renders the container', () => {
    const { container } = render(<Container>Container Component</Container>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
