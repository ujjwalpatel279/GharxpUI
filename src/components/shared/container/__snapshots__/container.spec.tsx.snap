// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders the container 1`] = `
<div
  class="c-container"
>
  Container Component
</div>
`;
